Helm3 v2.5.0

DavidMarchant opened the pull request:
Add helm3 support for v2.5

Tested on Rancher

DavidMarchant committed:
Add 2.5.0 Chart

Installs successfully.
Not complete, still to add are:
 - cleanup functionality
 - documentation

DavidMarchant committed:
Fix uninstall

Reduce bit-rot of cleanup template & fix for 2.5 namespace system
Alter default list of resources to cleanup, including specifying which
ns each is in

DavidMarchant committed:
Add docs

mhmxs commented on charts/storageos-operator/templates/storageoscluster_cr.yaml:
```diff
@@ -4,12 +4,12 @@ apiVersion: storageos.com/v1
-  namespace: {{ .Release.Namespace }}
-spec:
   namespace: {{ .Values.cluster.namespace }}
+spec:
```
I have to add this line to be able to deploy the stuff:
```
secretRefNamespace: {{ .Values.cluster.namespace }}
```

DavidMarchant committed:
Remove unused value

DavidMarchant committed:
Add additional guard before creating namespace

Attempted a simple use of 'lookup' for prexisting namespaces but
it doesn't find resources created by the release itself

aeroniero33 commented:
We need to bump the chart version for the CI to work

aeroniero33 commented:
Also the helm delete fails when the chart is installed in kube-system, only the operator gets deleted

DavidMarchant committed:
Revamp cleanup

Firstly, remove the too-optimistic use of YAML anchors in the values
file, they're not properly replaced when a user specifies a partial yaml
file to override some values.

Also, move the cleanup template to use two sections of hooks
pre-delete, which:
 - delete the storageoscluster CR
 - wait for it's pods to come down (to prevent the operator being
   deleted before it can do its job vis-a-vis resource deletion)
 and post-delete, which
 - delete 4 misc operator resources (CMs & secrets) that were being
   left behind after operator deletion
Then let helm delete all the resources that it created.

DavidMarchant committed:
Improve NOTES.txt

DavidMarchant committed:
Bump chart version

DavidMarchant committed:
Merge pull request #130 from storageos/v2.5.0-temp

V2.5.0 temp

angelos-p reviewed (approved):
LGTM

aeroniero33 commented:
We need to add automated tests for this however

DavidMarchant commented on charts/storageos-operator/Chart.yaml:
```diff
@@ -1,5 +1,5 @@
+appVersion: "v2.5.0"
 description: Cloud Native storage for containers
 name: storageos-operator
 version: 0.4.7
```
Think you're looking at an old version there mate! Has been bumped to 0.5.0
